### PR TITLE
MINOR: Ensure that DisplayName is set in all parameterized tests

### PR DESCRIPTION
**Pull request by dajac**
This is a follow-up to https://github.com/apache/kafka/pull/14687 as we found out that some parameterized tests do not include the test method name in their name. For the context, the JUnit XML report does not include the name of the method by default but only rely on the display name provided.

Example:
Screenshot 2023-11-28 at 16 55 36

The `testDescribeQuorumReplicationSuccessful` part was not there before.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/kafka/utils/TestInfoUtils.scala [hunk 1]
```diff
@@ -51,7 +51,7 @@ object TestInfoUtils {
       testInfo.getDisplayName().contains("quorum=zkMigration")
     }
   }
-  final val TestWithParameterizedQuorumName = "{displayName}.quorum={0}"
+  final val TestWithParameterizedQuorumName = "{displayName}.{argumentsWithNames}"
```
Comment (dajac) [inline core/src/test/scala/kafka/utils/TestInfoUtils.scala:55]: Changing this one to be aligned with https://github.com/apache/kafka/pull/14687.

File: tools/src/test/java/org/apache/kafka/tools/ToolsTestUtils.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
 
 public class ToolsTestUtils {
     /** @see TestInfoUtils#TestWithParameterizedQuorumName()  */
-    public static final String TEST_WITH_PARAMETERIZED_QUORUM_NAME = "{displayName}.quorum={0}";
+    public static final String TEST_WITH_PARAMETERIZED_QUORUM_NAME = "{displayName}.{argumentsWithNames}";
```
Comment (dajac) [inline tools/src/test/java/org/apache/kafka/tools/ToolsTestUtils.java:45]: Changing this one to be aligned with https://github.com/apache/kafka/pull/14687.

**Comment by mumrah** [comment]
I tried this in IntelliJ and I see the test name included in the individual test variation output. Looks great!

image

**Review by mumrah — APPROVED** [review]
LGTM, thanks @dajac!

**Pull request merged**